src/csharp/Grpc.Core/Internal/NativeExtension.cs

```diff
@@ -97,21 +94,31 @@ public NativeMethods NativeMethods
         /// </summary>
         private static UnmanagedLibrary Load()
         {
-            // TODO: allow customizing path to native extension (possibly through exposing a GrpcEnvironment property).
-
-            var libraryFlavor = string.Format("{0}_{1}", GetPlatformString(), GetArchitectureString());
-
-            var assemblyDirectory = Path.GetDirectoryName(GetAssemblyPath());
-
-            // With old-style VS projects, the native libraries get copied using a .targets rule to the build output folder
-            // alongside the compiled assembly.
-            var classicPath = Path.Combine(assemblyDirectory, NativeLibrariesDir, libraryFlavor, GetNativeLibraryFilename());
-
-            // DNX-style project.json projects will use Grpc.Core assembly directly in the location where it got restored
-            // by nuget. We locate the native libraries based on known structure of Grpc.Core nuget package.
-            var dnxStylePath = Path.Combine(assemblyDirectory, DnxStyleNativeLibrariesDir, libraryFlavor, GetNativeLibraryFilename());
-
-            return new UnmanagedLibrary(new string[] {classicPath, dnxStylePath});
+            string[] paths;
+            string explicitPath = GrpcEnvironment.NativeLibrary;
+            if (explicitPath != null)
+            {
+                paths = new[] { explicitPath };
+            }
+            else
+            {
+                var assemblyDirectory = Path.GetDirectoryName(GetAssemblyPath());
+
+                // With old-style VS projects, the native libraries get copied using a .targets rule to the build output folder
+                // alongside the compiled assembly.
+                // With dotnet cli projects, the native libraries (just the required ones) are similarly copied to the built output folder,
```

Comment:
Is this why the the shared library sources are now kept in ".../runtime/<os>-<arch>/..." instead of ".../build/..." directories? Is this following a dotnet convention of where it finds the required library?